assets/js/theme/common/product-details.js

```diff
@@ -271,17 +271,38 @@ export default class ProductDetails {
         if (_.isPlainObject(image)) {
             const zoomImageUrl = utils.tools.image.getSrc(
                 image.data,
-                this.context.themeSettings.zoom_size,
+                this.context.themeSettings.srcset_size_1920,
+                // Should match zoom size used for data-zoom-image in components/products/product-view.html
             );
 
             const mainImageUrl = utils.tools.image.getSrc(
                 image.data,
-                this.context.themeSettings.product_size,
+                this.context.themeSettings.srcset_size_640,
+                // Should match fallback image size used for the main
+                // product image in components/products/product-view.html
             );
 
+            /*
+                 Check the theme config looking for any keys containing `srcset_size_` and assuming them to
+                 be image size strings (e.g. "123x123") that will be used as sizes for srcset generation
+             */
+            const srcsets = Object.keys(this.context.themeSettings)
```

Comment:
this is brilliant. it's a shame themeSettings doesn't support an Array natively.

Reply:
Yep, this was just me getting it working with existing functionality... presumably we could add an array in the theme settings to make this cleaner. It wouldn't be editable in Theme Editor without more work on our end (since it's basically a new data type)

Reply:
I've moved away from using theme config, and instead moved the logic into a new helper and stencil-utils method. We'll have a strong set of default sizes, but allow configuration by theme developers.